### ISSUE TITLE
Tacotron-2 renew

### ISSUE THREAD

**Issue by dathudeptrai** [TensorSpeech/TensorFlowTTS#212]
Hmm, there is something wrong in my tacotron-2 implementation that can't apply mixed precision right now. About training speed, compared with nvidia tacotron-2 with my implementation with the same setting on both LJSpeech, 2080Ti, Cuda 10.1, Cudnn 7.6.5, r = 1, my training speed is 3.5s/1it while nvidia tacotron2 pytorch implementation is 3s/1it. In my private dataset, my implementation  convergence faster than nvidia implementation (let say just need 100k to get the same loss and quality with nvidia at 200k steps). So i think it's ok since the training speed 15% slower but 2x faster convergence. But that is float32, nvidia can use mixed precision to speed-up training and larger batch-size while my implemen can't apply mixed precision :(. I will re-implementation tacotron-2 in the difference way to make it able to use mixed_precision. Use mixed precision we can speed-up tacotron-2 training progress around 2x, that mean around 1.5-1.7s/1it. It's great if some one can work with me to make mixed precision available :D.

**Comment by jun-danieloh** [comment]
Hi @dathudeptrai Is this implementation on going now?

**Comment by dathudeptrai** [comment]
@jun-danieloh mixed precistion training is tensorflow bug, do not know if the next version of tensorflow fix it. The training speed now is comparable with pytorch float32 version.

**Comment by jun-danieloh** [comment]
@dathudeptrai How did you decide whether the training speed is compared to PyTorch? If it is true, what was your change? Can you point me to your commit of changes or something?

**Comment by dathudeptrai** [comment]
@dathudeptrai i used both my implementation and other pytorch implementation to training tacotron2 now, the training speed in LJSpeech of pytorch float32 is 2.8-> 3.0s while my implementaioni is 3.0-3.1. I think it's comparable. TF2.3 is faster than 2.2

**Comment by jun-danieloh** [comment]
@dathudeptrai Can we switch this implementation to TF2.3 without any changes? I am using TF2.2 now.

**Comment by dathudeptrai** [comment]
@jun-danieloh yes you can use tf 2.3 for training :D. Note that if you want to freeze the model to Pb, you need use 2.2 :)).

**Comment by jun-danieloh** [comment]
@dathudeptrai Do you know or remember how much training speed gain is when TF2.3 used?

**Comment by dathudeptrai** [comment]
> @dathudeptrai Do you know or remember how much training speed gain is when TF2.3 used?

3.5 -> 3.0 s.

**Comment by jun-danieloh** [comment]
@dathudeptrai My PyTorch implementation takes ~3 mins per an epoch(batch 128) but this tf2.2 takes ~44 mins(batch 24) so am still trying to figure out why this is so slow compared to PyTorch's one. My GPU is 1080Ti, 12G.

**Comment by jun-danieloh** [comment]
800 mel_lengths mins how much in second in wav files? I guess it is 8 second? Mine is also using fixed length and max_mel_length is set 2000 and it still shows same performance.

**Comment by dathudeptrai** [comment]
> 800 mel_lengths mins how much in second in wav files? I guess it is 8 second? Mine is also using fixed length and max_mel_length is set 2000 and it still shows the same performance.

max mel length should be <= 900, i think 800 is ok. 800 mel is (800 * hop_size) / hz (second). So if hop_size is 256, hz is 22050 in Ljspeech then we have 800 * 256 / 22050 = 9.28s.

**Comment by jun-danieloh** [comment]
@dathudeptrai Is this issue still not resolved? I just tried to train tacotron2 with mixed_precision setting but got errors.

**Comment by dathudeptrai** [comment]
> @dathudeptrai Is this issue still not resolved? I just tried to train tacotron2 with mixed_precision setting but got errors.

that is a tensorflow bug :)), I can't fix by myself :D .
